### PR TITLE
Add more informative error if shots argument is of wrong type

### PR DESCRIPTION
**Pull request by siddharthdangwal**
### Summary

fixes issue #4878 

### Details and comments

provides an informative error message to make the 'shots' argument in execute function of type 'int'.

### CODE REVIEW

File: qiskit/circuit/quantumcircuit.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ class QuantumCircuit:
     def __init__(self, *regs, name=None):
         if any([not isinstance(reg, (QuantumRegister, ClassicalRegister)) for reg in regs]):
             try:
+                if any([isinstance(reg, float) for reg in regs]):
```
Comment (taalexander) [inline qiskit/circuit/quantumcircuit.py:142]: Why is this change included in this PR?

File: qiskit/compiler/assemble.py [hunk 2]
```diff
@@ -220,10 +220,12 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             shots = min(1024, max_shots)
         else:
             shots = 1024
-    elif max_shots and max_shots < shots:
+    elif isinstance(shots, int) and max_shots and max_shots < shots:
         raise QiskitError(
             'Number of shots specified: %s exceeds max_shots property of the '
             'backend: %s.' % (shots, max_shots))
+    elif not isinstance(shots, int):
+        raise TypeError('The attribute \'shots\' must be of type int')
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:228]: This should raise a QiskitError. A note should be added in the Raises section that this may raise if shots is not right.

File: qiskit/compiler/assemble.py [hunk 3]
```diff
@@ -220,10 +220,12 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             shots = min(1024, max_shots)
         else:
             shots = 1024
-    elif max_shots and max_shots < shots:
+    elif isinstance(shots, int) and max_shots and max_shots < shots:
         raise QiskitError(
             'Number of shots specified: %s exceeds max_shots property of the '
             'backend: %s.' % (shots, max_shots))
+    elif not isinstance(shots, int):
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:227]: This check should be placed before the above elif branch and the isinstance check should be removed from the branch above.

File: qiskit/compiler/assemble.py (hunk 2 above)
Comment (taalexander) [inline qiskit/compiler/assemble.py:228]:
```suggestion
        raise TypeError('The attribute "shots" must be of type int')
```

**Comment by 1ucian0** [comment]
ping @siddharthdangwal ? The status of this PR is in "changes requested" for some time. Are you still working on this?

**Comment by 1ucian0** [comment]
I'm closing the PR as stalled. Please @siddharthdangwal, before continue working on this, check in #4878 if somebody else is not working on it already.
